### PR TITLE
Fixups for odroidc2 example

### PR DESCRIPTION
**Pull request by mudler**
Add symlink to vmlinuz and reduce image size by running zypper cc.
It also override default layout (don't require COS_PERSISTENT) and disable requiring network on boot.
Drops ubuntu sample - it's not working, and green one it's enough, hooks the image creation to the ci

Next:
- Annotate generated docker images and publish them during release

Signed-off-by: Ettore Di Giacinto <edigiacinto@suse.com>

### CODE REVIEW

File: .github/workflows/examples_publish.yaml [hunk 1]
```diff
@@ -5,6 +5,7 @@ on:
   push:
     branches:
       - master
+      - odroid_c2
```
Comment (mudler) [inline .github/workflows/examples_publish.yaml:8]: Gotta remove this after it passes :)

File: .github/workflows/examples_publish.yaml [hunk 2]
```diff
@@ -68,4 +71,34 @@ jobs:
...
+             echo "COS_VERSION=$(cos_version)" >> $GITHUB_ENV
+      - name: Build Image
+        id: build_image
+        run: |
+          mkdir build
+          sudo ./images/arm-img-builder.sh --docker-image ${{ steps.prep.outputs.current }} build/cOS-example-${{ matrix.tag }}$COS_VERSION.img
+      - name: Upload Image
+        run: |
+          sudo tar cvf build.tar build
+          sudo luet util pack quay.io/costoolkit/examples:${{ matrix.tag }}$COS_VERSION.img build.tar image.tar
```
Comment (mudler) [inline .github/workflows/examples_publish.yaml:87]: This creates a docker image for the raw `img` file. I suspect those files are too big to be released over Github, so we might end up just distributing images that can be pulled locally

**Review by Itxaka — COMMENTED** [review]
looks good.

the artifact name seems to have a missing field `odroid-c2--image`
Cant the image be compressed somehow? maybe convert it to qcow? I know its not the same format but they are easily interchanged and can results in a much smaller image, with some docs on how to decompress it should be ok

**Comment by Itxaka** [comment]
Maybe we should call the workflow differently? Like this are device example images no? As to not mistake them with our own docker example images

File: .github/workflows/examples_publish.yaml (hunk 1 above)
Comment (mudler) [reply .github/workflows/examples_publish.yaml:8]: It passed here: https://github.com/rancher-sandbox/cOS-toolkit/runs/4314722836?check_suite_focus=true

File: examples/odroid-c2/files/system/oem/01_layout.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+name: "Immutable Layout"
+stages:
+  rootfs:
+    - if: '[ ! -f "/run/cos/recovery_mode" ]'
```
Comment (Itxaka) [inline examples/odroid-c2/files/system/oem/01_layout.yaml:4]: Maybe a comment here on `why` we are doing this would be good for future us.

**Review by Itxaka — APPROVED** [review]

**Comment by mudler** [comment]
> looks good.
> 
> the artifact name seems to have a missing field `odroid-c2--image` 

Where? I see it being uploaded as:

```
 build/cOS-example-odroid-c2-0.7.4-5-gb0a380c9.img.sha256
build/cOS-example-odroid-c2-0.7.4-5-gb0a380c9.img
```

> Cant the image be compressed somehow? maybe convert it to qcow? 

Why qcow? we want to provide images that can be directly used for SD card etc

> I know its not the same format but they are easily interchanged and can results in a much smaller image, with some docs on how to decompress it should be ok

We could apply xz, but I avoided as the last step of the release already compress them. Besides the container image which is being uploaded is compressed already (see ` odroid-c2-0.7.4-5-gb0a380c9.img ` in https://quay.io/repository/costoolkit/examples?tab=tags)

**Review by Itxaka — APPROVED** [review]

File: examples/odroid-c2/files/system/oem/01_layout.yaml (hunk 3 above)
Comment (mudler) [reply examples/odroid-c2/files/system/oem/01_layout.yaml:11]: Addressed in https://github.com/rancher-sandbox/cOS-toolkit/pull/885/commits/3ba3bf43dcb161304218e48129044cdf21c97997 !

**Comment by mudler** [comment]
No CI jobs will run for this as it runs just on master and its a new pipeline, merging :t-rex:

**Pull request merged**